src/webui/api/torrentscontroller.cpp

```diff
@@ -1475,6 +1475,26 @@ void TorrentsController::addTagsAction()
     }
 }
 
+void TorrentsController::setTagsAction()
+{
+    requireParams({u"hashes"_s, u"tags"_s});
+
+    const QStringList hashes {params()[u"hashes"_s].split(u'|', Qt::SkipEmptyParts)};
+    const QStringList tags {params()[u"tags"_s].split(u',', Qt::SkipEmptyParts)};
+
+    TagSet newTags {tags.begin(), tags.end()};
+    applyToTorrents(hashes, [&newTags](BitTorrent::Torrent *const torrent)
```

Comment:
```suggestion
    applyToTorrents(hashes, [newTags](BitTorrent::Torrent *const torrent)
```
Otherwise, every next torrent that has an tag from the new list will eat it up, preventing it from being applied to the rest of the torrents.